.github/workflows/docs.yml

```diff
@@ -0,0 +1,37 @@
+name: docs
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  docs:
+    name: docs
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+
+      - name: Build docs
+        uses: actions-rs/cargo@v1
+        env:
+          RUSTDOCFLAGS: "--enable-index-page -Zunstable-options"
+        with:
+          command: doc
+          args: --no-deps --workspace
```

Comment:
Don't we need to enable the `docs` feature like we do for `docs.rs`? Sample over here https://github.com/matrix-org/matrix-rust-sdk/blob/master/matrix_sdk/Cargo.toml#L15

Reply:
What does the feature do? 

Builds fine in my fork: https://jsparber.github.io/matrix-rust-sdk/

Reply:
It adds annotations for feature specific methods using [doc_cfg](https://github.com/matrix-org/matrix-rust-sdk/blob/master/matrix_sdk/src/client.rs#L1149). Here's an example of such an annotation https://github.com/matrix-org/matrix-rust-sdk/blob/master/matrix_sdk/src/client.rs#L1149.

I mean we don't strictly need those annotations but it would be weird if our docs differ from those on docs.rs. Also note that this requires to be built on nightly due to `doc_cfg` being only available on nightly.

Reply:
I enabled the `docs` feature